polar-core/src/polar.lalrpop

```diff
@@ -475,16 +519,18 @@ pub Pattern: Value = {
     <List<"Pattern">>,
 };
 
-pub Value: Value = {
-    <Number>,
-    <PolarString>,
-    <Boolean>,
-    <Variable>,
-    <DictionaryTerm>,
-    <BuiltinOperation>,
-    <RewrittenOperation>,
-    <Call>,
-    <List<"Term">>,
+
+pub Value: ValueOrLogical = {
+    <IsLogical<BuiltinOperation>>,
+    <IsAny<Boolean>>,
+    <IsAny<Variable>>,
+    <IsAny<Call>>,
```

Comment:
Why are these `IsAny`? Wouldn't a variable always be a value and a call always be a logical?

Reply:
Variable can be a boolean which can be queried. E.g. 
```
x = true and x
```
You might be right for `Call`, although I think we maybe support doing `f(x) = f(1)` to unify to `x = 1`?

Reply:
ah yeah, that call unify thing keeps coming up, I don't like it but I guess that's ok